docs/src/main/sphinx/optimizer/pushdown.rst

```diff
@@ -151,6 +151,65 @@ to perform the remaining query processing on a smaller amount of data.
 The specifics for the supported pushdown of table joins varies for each data
 source, and therefore for each connector.
 
+However, there are some generic conditions that must be met in order for a join
+to be pushed down:
+
+* join pushdown must be enabled
+* all predicates that are part of the join must be possible to be pushed down
+* none of the tables in the join must be empty
+* all tables in the join must be from the same catalog
+
+You can check if pushdown for a specific join is performed by looking at the
+:doc:`EXPLAIN plan </sql/explain>` of the query. If a join is successfully
+pushed down to the connector, the explain plan does **not** show that ``Join`` operator.
+The explain plan only shows the operations that are performed by Trino.
+
+As an example, we loaded the TPCH data set into a PostgreSQL database and then
+queried it using the PostgreSQL connector::
+
+  SELECT c.custkey, o.orderkey
+  FROM orders o JOIN customer c ON c.custkey = o.custkey;
+
+You can get the explain plan by prepending the above query with ``EXPLAIN``::
+
+    EXPLAIN
+    SELECT c.custkey, o.orderkey
+    FROM orders o JOIN customer c ON c.custkey = o.custkey;
+
+The explain plan for this query does not show any ``Join`` operator,
+as this operation is now performed by the connector. You can see the
+generated SQL to perform the join as part of the PostgreSQL ``TableScan``
+operator. This shows you that the pushdown was successful.
+
+.. code-block:: text
+
+ Fragment 0 [SINGLE]
```

Comment:
For the purposes of this doc maybe we can make use of `EXPLAIN (TYPE LOGICAL, FORMAT TEXT)` to get more condensed plans.

That way you can maybe also add a plan with pushdown disabled to contrast it against.

No strong opinions on this one, it's pretty good as is.

Reply:
I prefer to go with `EXPLAIN (TYPE DISTRIBUTED)` simply because whenever I work with users, that is what they always use and are more familiar with since it's the default.

Reply:
Agreed on the familiarity part. It'd be confusing for most users.